### DISCUSSION TITLE
pip version conflict

### DISCUSSION THREAD

**Discussion by mgoldenbe** [CloudBytes-Academy/CloudBytes.dev#124] [Q&A] [answered]
After following the instructions, I tried `pip install sphinx` and got:

```
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 584, in _build_master
    ws.require(__requires__)
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 901, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 792, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.VersionConflict: (pip 22.2.1 (/usr/local/lib/python3.10/dist-packages), Requirement.parse('pip==20.0.2'))
```

**Comment by rehanhaider** [comment]
1. Converted this to discussion as this is not an issue with the installation.

**Comment by rehanhaider** [comment, marked as answer]
> ```
> pkg_resources.VersionConflict: (pip 22.2.1 (/usr/local/lib/python3.10/dist-packages), Requirement.parse('pip==20.0.2'))
> ```

This means that Sphinx is outdated and hasn't been updated to handle latest `pip` version. You might want to downgrade the version of the pip by running 

```bash
python3 -m pip install --upgrade pip==20.0.2
```
Then try to install sphinx again.

**Reply by mgoldenbe** [reply]
Thank you for the reply. I apologize, but I will not check your suggestion as I have already upgraded to Ubuntu 22.04, which has Python 10 to begin with.
